iree/turbine/kernel/wave/wave.py

```diff
@@ -321,6 +321,130 @@ def initialize_symbolic_constraints(self, trace: CapturedTrace) -> None:
                     subs_idxc(constraint.source_to_target(constraint.target)),
                 )
 
+    def infer_grid_shape(self, idxc: IndexingContext):
+        self.grid_type.dims = [1, 1, 1]
+        max_workgroup_dim = 2
+        aliases = [x.source for x in self.constraints if isinstance(x, SymbolicAlias)]
+        for constraint in self.workgroup_constraints:
+            if constraint.dim in aliases:
+                continue
+            if not constraint.primary:
+                continue
+            dim = (
+                constraint.workgroup_dim
+                if constraint.workgroup_dim < max_workgroup_dim
+                else max_workgroup_dim
+            )
+            self.grid_type.dims[dim] *= safe_subs(constraint.count, idxc.subs)
+
+    def try_apply_pass(
+        self,
+        p,
+        trace: CapturedTrace,
+        print_ir_before: Sequence[str] = [],
+        print_ir_after: Sequence[str] = [],
+    ):
+        if "all" in print_ir_before or p.__name__ in print_ir_before:
+            print(f"***Before {p.__name__}***\n")
+            print_trace(trace)
+        try:
+            p()
+        except Exception:
+            print(f"Error in pass: {p.__name__}\n")
+            print_trace(trace)
+            raise
+        if "all" in print_ir_after or p.__name__ in print_ir_after:
+            print(f"***After {p.__name__}***\n")
+            print_trace(trace)
+
+    def compile_to_scf_dialect(
```

Comment:
Not sure if `compile_to_scf_dialect` is a good name, probably just `compile_to_mlir`

Reply:
You reverted it back with last rebase I think

Reply:
you're right, I force pushed to get rid of the merges that I introduced clicking on the UI, fixing ..